lib/ws_servers/api/start_connections.js

```diff
@@ -75,21 +74,32 @@ class ConnectionManager {
 
   /**
    * @private
-   * @returns {Promise<void>}
    */
-  _startAlgoWorker ({ ws, apiKey, apiSecret }) {
-    return ws.algoWorker.start({
-      apiKey,
-      apiSecret,
-      userId: DEFAULT_USER
-    })
+  async _startAlgoWorker ({ server, ws, apiKey, apiSecret, mode, userId = DEFAULT_USER }, shouldUpdate = false) {
+    let algoWorker = ws.workers[mode]
+    if (algoWorker && algoWorker.isOpen && !shouldUpdate) {
```

Comment:
`shouldUpdate` will always be true when the user toggles between the modes, no? Is there a purpose of storing the existing worker instance in `ws.workers[mode]`

Reply:
done 4b4d97f20c99113e13859a7b8463bb3ed698763d